src/NBitcoin/ChainedBlock.cs

```diff
@@ -62,11 +64,30 @@ public ChainedBlock(BlockHeader header, uint256 headerHash, ChainedBlock previou
             this.CalculateChainWork();
         }
 
+        /// <summary>
+        /// Constructs a chained block.
+        /// </summary>
+        /// <param name="header">The header for the chained block.</param>
+        /// <param name="height">The height of the chained block</param>
+        public ChainedBlock(BlockHeader header, int height)
+        {
+            this.Header = header ?? throw new ArgumentNullException("header");
```

Comment:
Mike, changes this to:

`Guard.NotNull(header , nameof(header ));`

Reply:
This is in NBitcoin - no reference to Guard in here.

Reply:
good point :D

Reply:
Ah sorry of course!